nmrfx-processor/src/main/java/org/nmrfx/processor/processing/Processor.java

```diff
@@ -833,50 +828,59 @@ private int mapToFID(int i) {
         return mapToFID[i];
     }
 
-    private boolean useMemoryMode(int[] datasetSizes) {
+    public static boolean useMemoryMode(int[] datasetSizes) {
         long size = Float.BYTES;
         for (int i = 0; i < datasetSizes.length; i++) {
             size *= datasetSizes[i];
         }
-        boolean memoryMode = size < 135e6;
-        return memoryMode;
+        return useMemoryMode(size);
+    }
+
+    public static boolean useMemoryMode(long size) {
+        return size <=  MEMORY_MODE_LIMIT;
+    }
+
+    // used from Python for testing
+    public static void setMemoryModeLimit(long size) {
+        MEMORY_MODE_LIMIT = size;
     }
 
-    public boolean createNV(String outputFile, int datasetSizes[], int[] useSizes) {
-        boolean memoryMode = useMemoryMode(datasetSizes);
-        createNV(outputFile, datasetSizes, useSizes, memoryMode);
+    public boolean createNV(String outputFile, int[] useSizes) {
+        boolean memoryMode = useMemoryMode(useSizes);
+        createNV(outputFile, useSizes, memoryMode);
         return true;
     }
 
-    public boolean createNV(String outputFile, int datasetSizes[], int[] useSizes, Map flags) {
-        boolean memoryMode = useMemoryMode(datasetSizes);
-        createNV(outputFile, datasetSizes, useSizes, memoryMode);
+    public boolean createNV(String outputFile, int[] useSizes, Map flags) {
+        boolean memoryMode = useMemoryMode(useSizes);
+        createNV(outputFile, useSizes, memoryMode);
         setFidFlags(flags);
         return true;
     }
 
-    public boolean createNVInMemory(String outputFile, int datasetSizes[], int[] useSizes) {
-        createNV(outputFile, datasetSizes, useSizes, true);
+    public boolean createNVInMemory(String outputFile, int[] useSizes) {
+        createNV(outputFile, useSizes, true);
         return true;
     }
 
-    public boolean createNV(String outputFile, int datasetSizes[], int[] useSizes, boolean inMemory) {
+    public boolean createNV(String outputFile, int[] useSizes, boolean inMemory) {
         if (progressUpdater != null) {
             progressUpdater.updateStatus("Create output dataset");
         }
-        this.datasetSizes = datasetSizes;
-        this.acqSizesToUse = useSizes;  // fixme
-        if (nDim > datasetSizes.length) {
-            if (useSizes == null) {
-                log.error("specify useSizes if not using all dimensions");
-                return false;
-            }
+        int nDimToUse = 0;
+        for (var sz : useSizes) {
+            nDimToUse += sz > 1 ? 1 : 0;
         }
+        this.acqSizesToUse = useSizes;  // fixme
```

Comment:
perhaps a more descriptive comment about what exactly needs to be fixed here, otherwise this comment may stay here for years and eventually be forgotten what the problem was